hs_core/hydroshare/resource.py

```diff
@@ -1235,3 +1236,23 @@ def delete_metadata_element(resource_short_id, element_model_name, element_id):
     """
     res = utils.get_resource_by_shortkey(resource_short_id)
     res.metadata.delete_element(element_model_name, element_id)
+
+
+def save_resource_metadata_json(resource):
+    """
+    Writes the resource level metadata from the db in schema.org format to the hs_user_metadata.json file of a resource
+
+    Parameters:
+    :param resource: A resource instance
+    """
+    istorage = resource.get_s3_storage()
+    temp_path = utils.create_temp_dir_on_s3(istorage)
```

Comment:
Why create a temp directory on s3? It looks like you're writing a temporary local file before pushing it to the s3 resource.